pkg/kubelet/volumemanager/reconciler/reconciler_test.go

```diff
@@ -60,6 +60,40 @@ const (
 
 func hasAddedPods() bool { return true }
 
+func Test_UpdateStates(t *testing.T) {
+	// Arrange
+	volumePluginMgr, _ := volumetesting.GetTestVolumePluginMgr(t)
+	dsw := cache.NewDesiredStateOfWorld(volumePluginMgr)
+	asw := cache.NewActualStateOfWorld(nodeName, volumePluginMgr)
+	kubeClient := createTestClient()
+	fakeRecorder := &record.FakeRecorder{}
+	fakeHandler := volumetesting.NewBlockVolumePathHandler()
+	oex := operationexecutor.NewOperationExecutor(operationexecutor.NewOperationGenerator(
+		kubeClient,
+		volumePluginMgr,
+		fakeRecorder,
+		false, /* checkNodeCapabilitiesBeforeMount */
+		fakeHandler,
+	))
+	recon := NewReconciler(
+		kubeClient,
+		false, /* controllerAttachDetachEnabled */
+		reconcilerLoopSleepDuration,
+		waitForAttachTimeout,
+		nodeName,
+		dsw,
+		asw,
+		hasAddedPods,
+		oex,
+		mount.NewFakeMounter(nil),
+		hostutil.NewFakeHostUtil(nil),
+		volumePluginMgr,
+		kubeletPodsDir)
+	rec := recon.(*reconciler)
+	rec.kubeClient = nil
+	rec.updateDevicePath(nil)
```

Comment:
This isn't really testing the reconstruction path, but I guess we currently don't have any unit tests for reconstruction? @jingxu97 do you have any ideas?

Reply:
Yeah - trying to find good way to test it.

Reply:
@msau42 
Can the test be added in a follow-on PR ?